### ISSUE TITLE
Supporting Flysystem 2.x

### ISSUE THREAD

**Issue by frankdejonge** [Superbalist/flysystem-google-cloud-storage#111]
Hi all,

As I'm ramping up on the development of Flysystem 2.x (the initial alpha releases have been tagged) I was wondering if supporting it is on your road-map. I'm happy to put in the work as well, but I also don't want to swoop in and take the work you all have been putting in by creating this community supported adapter.

Let's start the conversation about how to get Google Cloud Storage available for Flysystem 2.x, I'm open to input.

Regards,

Frank

**Comment by nikolaos-spyratos** [comment]
Hey Frank,

Thanks for reaching out! We're keen to support Flysystem 2.x, but to be perfectly honest we haven't yet looked at how extensive that support would need to be. I'm assuming we just need to take a look at the [upgrade guide](https://flysystem.thephpleague.com/v2/docs/advanced/upgrade-to-2.0.0/) and [new features[(https://flysystem.thephpleague.com/v2/docs/what-is-new/)?

**Comment by frankdejonge** [comment]
Hi @nikolaos-spyratos,

What I did for the other implementations was: Create a new 2.x implementation, added all the empty methods (through PhpStorm), created a test case that extends the abstract test case (available as a package: `league/flysystem-adapter-test-utilities`), and filled in any missing cases. I copied over pieces of the old adapter internals and adjusted per case when needed. Although this sounds like a bit of work, it was quite easy and very time-consuming.
